src/Microsoft.Diagnostics.NETCore.Client/DiagnosticsIpc/IpcTransport.cs

```diff
@@ -100,6 +145,51 @@ public override int GetHashCode()
         }
     }
 
+    internal class AddressIpcEndpoint : IpcEndpoint
+    {
+        string _address;
+
+        public AddressIpcEndpoint(string address)
+        {
+            _address = address;
+        }
+
+        public override Stream Connect(TimeSpan timeout)
+        {
+            return IpcEndpointHelper.Connect(_address, timeout);
+        }
+
+        public override async Task<Stream> ConnectAsync(CancellationToken token)
+        {
+            return await IpcEndpointHelper.ConnectAsync(_address, token).ConfigureAwait(false);
+        }
+
+        public override void WaitForConnection(TimeSpan timeout)
+        {
+            using var _ = Connect(timeout);
+        }
+
+        public override async Task WaitForConnectionAsync(CancellationToken token)
+        {
+            using var _ = await ConnectAsync(token).ConfigureAwait(false);
+        }
+
+        public override bool Equals(object obj)
+        {
+            return Equals(obj as AddressIpcEndpoint);
+        }
+
+        public bool Equals(AddressIpcEndpoint other)
+        {
+            return other != null && other._address == _address;
+        }
+
+        public override int GetHashCode()
+        {
+            return _address.GetHashCode();
+        }
+    }
```

Comment:
Rather than have an `AddressIpcEndpoint`, I think it might be a cleaner abstraction to have all `IpcEndpoint` variants have an `IpcEndpointConfig` on them. I should be able to ask an `IpcEndpoint` what its address and schema are.

Reply:
I have implemented something inline with this.

Reply:
> I should be able to ask an IpcEndpoint what its address and schema are

Sorry I didn't get to this earlier. While I am not opposed if you guys already implemented it, I don't understand what scenario we would ever invoke such code if we had it? For some our IpcEndpoints these values probably aren't well-defined, for example IpcEndpoints corresponding to processes that connect to listener we created.

Reply:
I only implemented the ability to use an IpcEndpointConfig, inline with comments elsewhere, so calls to Connect/ConnectAsync can take a more generic IpcEndpointConfig instead of string address.

Reply:
I'm good with that